src/box/iproto.h

```diff
@@ -120,6 +136,15 @@ iproto_rmean_foreach(void *cb, void *cb_ctx);
 int
 iproto_thread_rmean_foreach(int thread_id, void *cb, void *cb_ctx);
 
+/**
+ * Sets an IPROTO request handler with the provided callback, destructor and
+ * context for the given request type.
+ * Returns 0 on success, a non-zero value on error (diagnostic is set).
+ */
+int
+iproto_override(uint32_t request_type, iproto_handler_f callback,
+		void (*destructor)(void *), void *ctx);
```

Comment:
Nit: It'd be neat to have a type defined for IPROTO handler destructor - for consistency with the callback.

Reply:
AFAIR, in the discussion about passing allocators to `box_session_ids` you said the opposite: why add redudant typedefs for such stuff — and referred to [`pthread_create`](https://man7.org/linux/man-pages/man3/pthread_create.3.html).

Reply:
Whether to add or not to add a callback apparently depends on the context. `alloc` and `free` are common functions with well-defined arguments so adding callbacks for them would only clutter the API. An override handler destructor OTOH is more complicated (What is this `void *` for? It's unclear from the declaration or the comment that it's `ctx`.) In this case it's better to add a typedef with a proper comment. Besides, you already added a typedef for the callback itself so now the function API looks lopsided.

Reply:
Sounds resonable, added a typedef.